Classes/Domain/Repository/CopyrightReferenceRepository.php

```diff
@@ -74,53 +76,76 @@ public function findByRootline($settings) {
         if(false === empty($finalRecords)) {
             $finalQuery = $this->createQuery();
             return $finalQuery->statement('SELECT * FROM sys_file_reference WHERE uid IN('.implode(',',$finalRecords).')')->execute();
-        } else {
-            return [];
         }
 
+        return [];
     }
 
     /**
      * @param string $rootlines
-     * @return array|\TYPO3\CMS\Extbase\Persistence\QueryResultInterface
+     * @return array
      */
     public function findForSitemap($rootlines) {
 
-        $pidClause = $this->getStatementDefaults($rootlines);
-
-        // First main statement, exclude by all possible exclusion reasons
-        $preQuery = $this->createQuery();
+        $context = GeneralUtility::makeInstance(\TYPO3\CMS\Core\Context\Context::class);
+        $sysLanguage = (int) $context->getPropertyFromAspect('language', 'id');
 
-        $now = time();
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('sys_file_reference');
 
-        // TODO: Migrate to QueryBuilder for Cross-DB-Engines
-        $statement = '
-          SELECT ref.* FROM sys_file_reference AS ref
-          LEFT JOIN sys_file AS file ON (file.uid=ref.uid_local)
-          LEFT JOIN pages AS p ON (ref.pid=p.uid)
-          WHERE p.deleted=0 AND p.hidden=0 AND (p.starttime=0 OR p.starttime<='.$now.') AND (p.endtime=0 OR p.endtime>='.$now.')
-          AND file.missing=0 AND file.uid IS NOT NULL AND (file.type=2 OR file.type=5)
-          AND ref.deleted=0 AND ref.hidden=0 AND ref.t3ver_wsid=0 '. $pidClause;
+        $constraints = [
+            $queryBuilder->expr()->eq('ref.sys_language_uid', $sysLanguage),
+            $queryBuilder->expr()->eq('missing', 0),
+            $queryBuilder->expr()->isNotNull('file.uid'),
+            $queryBuilder->expr()->in('file.type', [2, 5]),
+        ];
 
-        $preQuery->statement($statement);
+        if ('' !== $rootlines) {
+            $constraints[] = $queryBuilder->expr()->in('ref.pid', $this->extendPidListByChildren($rootlines));
+        }
 
-        $preResults = $preQuery->execute(TRUE);
+        $preResults = $queryBuilder
+            ->selectLiteral('ref.uid', 'ref.tablenames', 'ref.uid_foreign')
+            ->from('sys_file_reference', 'ref')
+            ->leftJoin(
+                'ref',
+                'sys_file',
+                'file',
+                $queryBuilder->expr()->eq('file.uid', 'ref.uid_local')
+            )
+            ->join(
```

Comment:
this changed to an actual join due a small side effect of the current implementation:

The deleted + hidden constraints that are added by the QueryBuilder will be added to the ON clause on a leftJoin.
That's because those fields would always be NULL in case there is no record to join thus the records affected would be removed because of the hidden = 0 constraint.

Basically the previous implementation was already a normal inner join even if it says LEFT JOIN